manifest/provider/import_test.go

```diff
@@ -0,0 +1,65 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestParseImportID(t *testing.T) {
+	samples := []struct {
+		ID        string
+		GVK       schema.GroupVersionKind
+		Name      string
+		Namespace string
+		Err       error
+	}{
+		{
+			ID:        "apiVersion=v1,kind=ConfigMap,namespace=default,name=test",
+			GVK:       schema.GroupVersionKind{Group: "", Version: "v1", Kind: "ConfigMap"},
+			Name:      "test",
+			Namespace: "default",
+			Err:       nil,
+		},
+		{
+			ID:        "apiVersion=rbac.authorization.k8s.io/v1,kind=ClusterRole,name=test",
+			GVK:       schema.GroupVersionKind{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: "ClusterRole"},
+			Name:      "test",
+			Namespace: "",
+			Err:       nil,
+		},
+		{
+			ID:        "apiVersion=apps/v1,kind=Deployment,namespace=foo,name=bar",
+			GVK:       schema.GroupVersionKind{Group: "apps", Version: "v1", Kind: "Deployment"},
+			Name:      "bar",
+			Namespace: "foo",
+			Err:       nil,
+		},
+		{
+			ID:  "foobar",
+			Err: fmt.Errorf("invalid format for import ID [%s]\nExpected format is: apiVersion=<value>,kind=<value>,name=<value>[,namespace=<value>]", "foobar"),
+		},
+	}
+	for _, s := range samples {
+		gotGvk, gotName, gotNamespace, gotErr := parseImportID(s.ID)
```

Comment:
We should rename `s` -> `expected` and then change the `got` prefix in each of these variables to `actual` – expected and actual are the usual names in our tests I think.